### PR TITLE
Cut Release 0.12

### PR DESCRIPTION
**Pull request by raptorsun**
## Description

This PR prepares for cutting release 0.12:

- Updated changelog
- Updated compatibility matrix
- Pinned dependencies

Once this PR is merged, the next steps are:
1. create a branch release-0.12. 
2. create a follow-up PR to restore branch settings in `jsonnetfile.json`.

## Type of change

_What type of changes does your code introduce to the kube-prometheus? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

### CODE REVIEW

**Comment by raptorsun** [comment]
PR to fix the broken document link: https://github.com/prometheus-operator/kube-prometheus/pull/1920

File: README.md [hunk 1]
```diff
@@ -40,12 +40,12 @@ no effect, but is still deployed.
...
-| [`release-0.10`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.10) | ✗               | ✗               | ✔               | ✔               | ✗               |
-| [`release-0.11`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.11) | ✗               | ✗               | ✗               | ✔               | ✔               |
+| [`release-0.9`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.9)   | ✔               | ✔               | ✗               | ✗               | ✗               |
+| [`release-0.10`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.10) | ✗               | ✔               | ✔               | ✗               | ✗               |
+| [`release-0.11`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.11) | ✗               | ✗               | ✔               | ✔               | ✗               |
+| [`release-0.12`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.11) | ✗               | ✗               | ✗               | ✔               | ✔               |
```
Comment (jan--f) [inline README.md:48]:
```suggestion
| [`release-0.12`](https://github.com/prometheus-operator/kube-prometheus/tree/release-0.12) | ✗               | ✗               | ✗               | ✔               | ✔               |
```

Comment (raptorsun) [reply README.md]: thanks, corrected now :)

**Comment by PhilipGough** [comment]
@raptorsun sorry for the delay on this. Do you mind rebasing and getting this in shape?

**Comment by raptorsun** [comment]
> @raptorsun sorry for the delay on this. Do you mind rebasing and getting this in shape?

no problem, I will update the PR soon :)

**Comment by raptorsun** [comment]
PR has been rebased and new changes are put into the change log. 
The branch for kubernetes-mixin is still master. When we have created release branch on kubernetes-mixin, I will update again the PR to pin that repo to release-0.12 branch. 
@PhilipGough 👋

**Comment by raptorsun** [comment]
All pre-requisites are ok. This PR is ready to review.

**Review by jan--f — APPROVED** [review]
LGTM :+1:

**Comment by simonpasquier** [comment]
@PhilipGough couldn't we create the `release-0.12` branch in advance and have this PR target the new branch? It would avoid the need of unpinning the dependencies in a subsequent PR.

**Comment by raptorsun** [comment]
> @PhilipGough couldn't we create the `release-0.12` branch in advance and have this PR target the new branch? It would avoid the need of unpinning the dependencies in a subsequent PR.

That's more simple, I wondered if we can merge this PR directly into a new release branch, too :)

**Comment by PhilipGough** [comment]
@simonpasquier yes, thats the process that should be followed according to the release notes so let me create a release branch and we can target that with this PR

**Comment by raptorsun** [comment]
Hello guys, the PR is ready to review and merge. 
The branch https://github.com/prometheus-operator/kube-prometheus/tree/release-0.12 has been created, so the test `Check Documentation formatting and links (pull_request)` should pass if we run it again.

**Comment by PhilipGough** [comment]
Thanks @raptorsun

**Review by PhilipGough — APPROVED** [review]
lgtm

**Pull request merged**